azure/services/agentpools/agentpools.go

```diff
@@ -53,6 +53,7 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 	profile := containerservice.AgentPool{
 		ManagedClusterAgentPoolProfileProperties: &containerservice.ManagedClusterAgentPoolProfileProperties{
 			VMSize:              containerservice.VMSizeTypes(agentPoolSpec.SKU),
+			OsType:              containerservice.Linux,
```

Comment:
in the future we should consider making that configurable and supporting Windows nodes as well

Reply:
Good point, I was thinking about that. Is Windows support working now in CAPZ for regular clusters?

Do you think it would work for AKS clusters if it was possible to configure it here?

I would prefer to have that in a separate PR, if you agree, and keep here this simple fix for the mentioned use case (as only Linux is supported ATM anyway).

Reply:
Adding Windows support might require other changes and additions to the current managed cluster implementation, which would increase PR scope, while the idea here was just to slightly improve current Linux support.

Reply:
AKS already supports Windows nodes today, it's just not exposed in CAPZ right now. I'm good with a follow up to implement windows, but can you please file an issue noting that Linux is currently hardcoded in the ManagedClusterAgentPoolProfileProperties?

Reply:
we also really need to do #611 so we can test this stuff in PRs

Reply:
Field an issue for Windows support for AKS https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1176. Also added a note there that Linux is currently hardcoded in the ManagedClusterAgentPoolProfileProperties.

> we also really need to do #611 so we can test this stuff in PRs

Yes, definitely, I mentioned adding e2e tests here https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1173 for the AKS cluster adoption scenario, which is currently not working due to few issues, including missing OS type in ManagedClusterAgentPoolProfileProperties.

Reply:
Looking into failed test now.

Reply:
Updated and fixed agent pool test.